### PR TITLE
add support for categories frontmatter

### LINKED ISSUE #129: support both category and categories front-matter
"out of the box" Jekyll supports both `category` and `categories`: https://jekyllrb.com/docs/frontmatter/#predefined-variables-for-posts

Could update the line [here](https://github.com/jekyll/jekyll-feed/blob/master/lib/feed.xml#L74) to support `categories` front matter.

### PR DESCRIPTION
**Pull request by pathawks**
Replaces #130

Fixes #129

### CODE REVIEW

**Comment by pathawks** [comment]
I had to create a second fixture site because the feed is limited to 10 items, and the feed already contained 10 items 😂

File: spec/jekyll-feed_spec.rb [hunk 1]
```diff
@@ -261,6 +261,33 @@
...
+      post = feed.items.last
+      expect(post.categories[0].term).to eql('single')
+      expect(post.categories.count).to eql(1)
+    end
+
+    it "outputs categories for posts with one `category`" do
+      post = feed.items.first
+      expect(post.categories[0].term).to eql('1')
+      expect(post.categories[1].term).to eql('2')
+      expect(post.categories[2].term).to eql('3')
```
Comment (pathawks) [inline spec/jekyll-feed_spec.rb]: Perhaps I should name the categories `first`, `second`, `third`?

**Comment by nebulousdog** [comment]
thanks @pathawks!

**Comment by danilowoz** [comment]
Please, someone, merge it 🙏

File: spec/jekyll-feed_spec.rb [hunk 2]
```diff
@@ -289,6 +289,33 @@
...
+      expect(feed.items.count).to eql(2)
+    end
+
+    it "outputs categories for posts with one `category`" do
+      post = feed.items.last
+      expect(post.categories[0].term).to eql('single')
+      expect(post.categories.count).to eql(1)
+    end
+
+    it "outputs categories for posts with one `category`" do
```
Comment (DirtyF) [inline spec/jekyll-feed_spec.rb]: @pathawks you meant multiples categories?

**Comment by nebulousdog** [comment]
thanks for pushing this forward all

**Comment by n1k0** [comment]
Do we have any news with this? Would be nice being able to support the `categories` front matter property; right now configuring `feed: categories:` in `_config.yml` will generate empty feeds for post categories using `categories` instead of `category`, which is super misleading to newcomers.

**Comment by SGOrava** [comment]
> Do we have any news with this? Would be nice being able to support the `categories` front matter property; right now configuring `feed: categories:` in `_config.yml` will generate empty feeds for post categories using `categories` instead of `category`, which is super misleading to newcomers.

Oh man, can you put this into README ?
This killed this plugin for me, unusable.

**Comment by ashmaroli** [comment]
@pathawks I pushed changes to how this was tested. Do you have any issues with the current diff..?

**Comment by jeffgable** [comment]
As @SGOrava mentioned, until this is merged and released, can you at least put it in the README that "categories" frontmatter is not supported?  Not knowing that just cost me two hours.

**Comment by DirtyF** [comment]
@pathawks What we should do here? 
We query only the posts from the `page.category`, then we loop on `post.categories`? what about the `page.category`?

**Comment by DirtyF** [comment]
@jekyllbot: merge +minor

**Comment by nebulousdog** [comment]
wonderful

**Pull request merged**